### PR TITLE
Add support for ephemeral_storage_config in container node_config

### PR DESCRIPTION
**Pull request by zapo**
Add support for `ephemeral_storage_config` in container/node_config

Fixes https://github.com/hashicorp/terraform-provider-google/issues/8461

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added field `ephemeral_storage_config` to resource `google_container_node_pool` and `google_container_cluster` (beta)
```

### CODE REVIEW

**Comment by zapo** [comment]
I wasn't able to run the acceptance tests for some reason using the provided documentation in the beta repo so hopefully that PR is not completely wrong! I must have skipped a step by mistake.
```
can't load package: package ./google: cannot find package "." in:
        /home/zapo/go/src/github.com/hashicorp/terraform-provider-google-beta/google
make: *** [GNUmakefile:15: testacc] Error 1
```

File: mmv1/third_party/terraform/utils/node_config.go.erb [hunk 1]
```diff
@@ -340,6 +359,13 @@ func expandNodeConfig(v interface{}) *containerBeta.NodeConfig {
 		nc.LocalSsdCount = int64(v.(int))
 	}
 
+	if v, ok := nodeConfig["ephemeral_storage_config"]; ok && len(v.([]interface{})) > 0 {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/utils/node_config.go.erb:362]: Can you beta guard this block as well?

File: mmv1/third_party/terraform/utils/node_config.go.erb [hunk 2]
```diff
@@ -571,6 +600,16 @@ func flattenShieldedInstanceConfig(c *containerBeta.ShieldedInstanceConfig) []ma
 	return result
 }
 
+func flattenEphemeralStorageConfig(c *containerBeta.EphemeralStorageConfig) []map[string]interface{} {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/utils/node_config.go.erb:603]: We'll also want to beta guard here

File: mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb [hunk 3]
```diff
@@ -849,6 +849,53 @@ func TestAccContainerNodePool_shieldedInstanceConfig(t *testing.T) {
 	})
 }
 
+<% unless version.nil? || version == 'ga' -%>
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb:852]:
```suggestion
<% unless version == 'ga' -%>
```

Beta guards don't need to check for `nil` anymore, version is always filled in (but we definitely have not backfilled all the code)

Comment (zapo) [reply mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb:852]: Thank you, I was confused about why in some places some of those where still checking for nil and just went with the style used in parts of tests.

File: mmv1/third_party/terraform/utils/node_config.go.erb (hunk 2 above)
Comment (zapo) [reply mmv1/third_party/terraform/utils/node_config.go.erb:606]: Updated

File: mmv1/third_party/terraform/utils/node_config.go.erb (hunk 1 above)
Comment (zapo) [reply mmv1/third_party/terraform/utils/node_config.go.erb:363]: Updated

**Comment by zapo** [comment]
@rileykarson Thanks for the quick turnaround on this PR.
I've pushed the fix for bad cast that I should have caught if I had tested this properly prior PR'ing. Seems like your test suite have caught it :)
I figured out how to test this out manually on a GCP project and was able to validate the SSD annotation was added properly on my nodes.
I'm not sure if your test suite actually deploys a real cluster internally but I fixed the acceptance test to represent the limitations of the ephemeral storage functionality which might help if that's the case.

**Comment by rileykarson** [comment]
Yeah, we deploy a real cluster. Don't worry about the history- we squash at merge time.

This test failed in an unexpected way- the cluster got stuck attempting to create. I'm going to look at it in more detail, but I won't have time for a few days. If I haven't posted an update on Thursday or Friday, mind pinging here? Thanks!

File: mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb [hunk 4]
```diff
@@ -849,6 +849,56 @@ func TestAccContainerNodePool_shieldedInstanceConfig(t *testing.T) {
...
+  location           = "us-central1-a"
+  initial_node_count = 1
+}
+
+resource "google_container_node_pool" "np" {
+  name               = "%s"
+  location           = "us-central1-a"
+  cluster            = google_container_cluster.cluster.name
+  initial_node_count = 1
+  min_master_version = "1.18"
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb:889]:
```suggestion
```

File: mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb [hunk 5]
```diff
@@ -849,6 +849,56 @@ func TestAccContainerNodePool_shieldedInstanceConfig(t *testing.T) {
...
+		},
+	})
+}
+
+func testAccContainerNodePool_ephemeralStorageConfig(cluster, np string) string {
+	return fmt.Sprintf(`
+resource "google_container_cluster" "cluster" {
+  name               = "%s"
+  location           = "us-central1-a"
+  initial_node_count = 1
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb:881]:
```suggestion
  initial_node_count = 1
  min_master_version = "1.18"
```

**Review by rileykarson — COMMENTED** [review]
I think `min_master_version` was misplaced, suggestions inline. I wonder if your config changes will make the test pass.

**Review by rileykarson — APPROVED** [review]

**Pull request merged**